ETL/Provision/provision_lambda.py

```diff
@@ -0,0 +1,71 @@
+"A script to read the url, and product data from the database"
+from os import environ as ENV
+import logging
+
+from dotenv import load_dotenv
+import psycopg2
+import psycopg2.extras
+from psycopg2.extensions import connection, cursor
+
+
+def get_connection(config) -> connection:
+    "Establishes a connection with the database"
+    return psycopg2.connect(
+        user=config["DB_USER"],
+        password=config["DB_PASSWORD"],
+        host=config["DB_HOST"],
+        port=config["DB_PORT"],
+        database=config["DB_NAME"]
+    )
+
+
+def get_cursor(conn: connection) -> cursor:
+    "Returns a cursor for the database"
+    return conn.cursor(cursor_factory=psycopg2.extras.RealDictCursor)
+
+
+def read_database(conn: connection):
+    "Gets the required data from the database"
+    with get_cursor(conn) as cur:
+        cur.execute("""SELECT DISTINCT ON (product_id) product_id, url, price
+                    FROM products
+                    LEFT JOIN price_readings USING (product_id)
+                    ORDER BY product_id, reading_at DESC""")
+        data = cur.fetchall()
+    logging.info("Product details taken from database")
+    return data
+
+
+def group_data(data: list[dict], processing_batch_size: int) -> list[list[dict]]:
```

Comment:
What happens if processing_bach_size is not of type int? It could be worth type checking this

Reply:
Added a check for this